### ISSUE TITLE
Flow feature extraction

### ISSUE THREAD

**Issue by junwenchen** [xumingze0308/TRN.pytorch#17]
Hi Mingze,

According to your LSTR paper, we found a kinetics pretrained model in mmaction2 library is used to extract motion features.
"Motion features are extracted from pre-computed stacked optical flow fields between 6 consecutive frames."

However, we found that the TSN motion extraction model in mmaction2 is pretrained using 5 consecutive frames, e.g. [model link](https://github.com/open-mmlab/mmaction2/blob/master/configs/recognition/tsn/tsn_r50_320p_1x1x8_110e_kinetics400_flow.py) 

Can you let me know which pretrained model you have used to extract flow features?

Many thanks.

**Comment by xumingze0308** [comment]
These are just alternative statements. In LSTR, we mean 6 consecutive **rgb** frames, which correspond to 5 consecutive **optical flow** frames.

**Comment by junwenchen** [comment]
Hi Mingze,

Could you please let me know what is the number of channels in each transformer decoder unit?

**Comment by xumingze0308** [comment]
We used 16.

**Comment by junwenchen** [comment]
Thanks.

What I mean is the number of the hidden size in the transformer decoder. Will it be so small?

**Comment by xumingze0308** [comment]
I see. The hidden size is 1024 and the number of heads is 16. I will clarify these hyperparameters later. Thanks!

**Comment by junwenchen** [comment]
Hi,

I just realize that the different ways of extracting labels will result in different accuracy.
Could you please share the feature extraction script with me?
If not convenient, can you share with me an example?, such as one of the label file video_validation_0000051.npy (of size L x 22)

Thanks.

**Comment by junwenchen** [comment]
Thanks for your reply.

"The different way" is that we have 24 frames per second and 6 frames per chunk. Then which frame's label is used as the chunk-wise label? How about the last chunk that might not have enough 6 frames? Will it be ignored or padded?
Currently I can almost reproduce the tendency of the curve in Fig.3. But the values on the curve are 2% lower than the paper. So I am wondering if that is due to feature extraction. 

Besides, could you please share a trained model or the test script, so that I can make sure the feature extraction is correct.
Is it possible to share the results of class-wise AP in LSTR?

Many thanks to your effort.

**Comment by junwenchen** [comment]
Btw, what is the initialization of any weights in the transformer decoders?

**Comment by xumingze0308** [comment]
The logic we used is as follows: (1) For action boundaries are labeled in seconds, we also convert them into frames-level with 4 FPS. For example, an action instance from 20.2s to 30.1s is from frame 80 (i.e., int(20.2 * 4)) to 120. (3) We annotate each chunk (with 6 frames) as a whole based on 4 FPS. I don't think there is a common rule for this, but it should not affect the performance too much as long as the training and testing are consistent. Using the center frame for chunk label is also reasonable to me. 

For the last chunk that has less than 6 frames, we just ignore this chunk.

We didn't apply any particular initialization for the transformers, and just trained from scratch.

**Comment by xumingze0308** [comment]
The logic we used is as follows: (1) For action boundaries are labeled in seconds, we convert them into frames-level with 4 FPS. For example, an action instance from 20.2s to 30.1s is from frame 80 (i.e., int(20.2 * 4)) to 120. (2) We annotate each chunk (with 6 frames) as a whole based on 4 FPS. I don't think there is a common rule for this, but it should not affect the performance too much as long as the training and testing are consistent. Using the center frame for chunk label is also reasonable to me.

For the last chunk that has less than 6 frames, we just ignore this chunk.

We didn't apply any particular initialization for the transformers, and just trained from scratch.

**Comment by junwenchen** [comment]
Thanks for response.

I have one more question related to TVSeries dataset. I found that in data_info.json file that the TVseries dataset does not have background class while the other two have. So how did you process the background frames in TVsereis?

**Comment by xumingze0308** [comment]
We should have the "background class" for TVSeries. That data info could be outdated.

**Comment by junwenchen** [comment]
Great.

Can you let me know the exact size of the model?

**Comment by junwenchen** [comment]
Hi,

I have one more question.
What is the total number of samples in the training dataloader?

many thanks

**Comment by xumingze0308** [comment]
The code of LSTR is released here: https://github.com/amazon-research/long-short-term-transformer.
